### PR TITLE
Enhance views

### PR DESCRIPTION
**Pull request by brunvez**
Refactor views using Tailwind

- [x] Sign in
- [x] Sign Up
- [x] Forgot password
- [ ] Change Password

image

### CODE REVIEW

**Comment by brunvez** [comment]
Just a concept on one way to organize views. The ones I've built are low-level components but we can easily compose. I also suck at designing so feel free to criticize 😅 .

The alternative I see to this is to add helpers but I feel that's just doing components in the wrong way.

File: Gemfile [hunk 1]
```diff
@@ -29,6 +29,8 @@ gem 'puma', '~> 5.0'
 gem 'sass-rails', '>= 6'
 # An utility first CSS framework
 gem 'tailwindcss-rails', '~> 0.3.0'
+# A framework for building reusable, testable & encapsulated view components in Ruby on Rails
+gem 'view_component', '~> 2.24', require: 'view_component/engine'
```
Comment (santib) [inline Gemfile:33]: I'm not against `view_component` but I'm worried to continue to add dependencies that will make it harder for people to get started with the base.. thoughts?

File: app/components/button_component.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      rounded-lg
+      shadow-md
+      hover:bg-gray-700
+      focus:outline-none
+      focus:ring-2
+      focus:ring-black
+      focus:ring-offset-2
+      focus:ring-offset-purple-200
+      cursor-pointer
+    ]
```
Comment (santib) [inline app/components/button_component.rb:35]: 😅

File: Gemfile (hunk 1 above)
Comment (JuannFerrari) [reply Gemfile:33]: I don't like using more things than the default rails form helpers + CSS classes usually. But in this case with so many utility classes (thanks to Tailwind) could be a good way to make reusable components and make the UI more uniform

Comment (brunvez) [reply Gemfile:33]: I think this is more like "this is the way we do it", like a live playbook. I really don't like dependencies but in this case I think Tailwind was designed to be used with components, I don't see it as any other way. I've also thought of these while building this PR:

```ruby
# erb
<%= f.submit 'Sign in', class: button_css %>

# helper
module ButtonsHelper
  def button_css
    %w[....]
  end
end
```

And

```ruby
# erb
<%= button('Sign in') %>

# helper
module ButtonsHelper
  def button
    button_tag('Sign in', class: %w[....])
  end
end
```

I'm afraid the latter would collide with Rails' form helpers and that it's actually just components done in a functional manner

Comment (santib) [reply Gemfile:33]: > I really don't like dependencies but in this case I think Tailwind was designed to be used with components, I don't see it as any other way

@brunvez I think Tailwind would work just fine with plain Rails partials

Comment (brunvez) [reply Gemfile:33]: Yeah but how would you extract common css? I'm alright sprinkling. utility classes here and there, but when they are as many as the button have I'd rather keep them in one place away from the rest of the code

Comment (santib) [reply Gemfile:33]: you can have a shared button partial such as you created a button component, right?

**Comment by brunvez** [comment]
As discussed previously we'll use ViewComponent only to do base components and keep using partials for specific views

**Pull request merged**